src/server/main_service.cc

```diff
@@ -1047,6 +1047,30 @@ void Service::DispatchCommand(CmdArgList args, facade::ConnectionContext* cntx)
   }
 }
 
+class ReplyGuard {
+ public:
+  ReplyGuard(ConnectionContext* cntx, std::string_view cid_name) {
+    auto* builder = cntx->reply_builder();
+    const bool is_script = bool(cntx->conn_state.script_info);
+    const bool is_one_of =
+        absl::flat_hash_set<std::string_view>({"REPLCONF", "DFLY", "EXEC", "EVALSHA"})
+            .contains(cid_name);
+    const bool should_dcheck = !is_one_of && !is_script;
+    if (should_dcheck) {
+      builder->ExpectReply();
```

Comment:
I do not understand this logic.

Reply:
It should have been `false`. Basically, it sets that we expect the reply on each `Invoke` which internally sets the `has_replied_` to false. If the reply builder is used to reply, then this is set to true, which we `DCHECK` on destructor